app/uk/gov/hmrc/thirdpartyapplication/services/ApplicationService.scala

```diff
@@ -389,32 +388,22 @@ class ApplicationService @Inject()(applicationRepository: ApplicationRepository,
       }
     }
 
-    def createAppData(ids: Option[TotpId]): ApplicationData = {
-      def newPrivilegedAccess = {
-        application.access.asInstanceOf[Privileged].copy(totpIds = ids)
-      }
-
-      val updatedApplication = ids match {
-        case Some(_) if application.access.accessType == PRIVILEGED => 
-          application match {
-            case v1 : CreateApplicationRequestV1 => v1.copy(access = newPrivilegedAccess)
-            case v2 : CreateApplicationRequestV2 => v2.copy(access = newPrivilegedAccess)
-          }
-        case _ => application
+    def applyTotpForPrivAppsOnly(totp: Option[Totp], request: CreateApplicationRequest): CreateApplicationRequest = {
+      request match {
+        case v1 @ CreateApplicationRequestV1(_, priv: Privileged, _,_,_,_) => v1.copy(access = priv.copy(totpIds = extractTotpId(totp)))
```

Comment:
is there a reason for using '@' here rather than ':' ?

Reply:
@ for values, : for types (so priv is matching an object of a type and is : but v1 is matching a pattern so @) is the best way I can describe it.